docs/source/configuration/external_timelock_service_configs/timelock_server_config.rst

```diff
@@ -0,0 +1,136 @@
+
+This is done for performance reasons: consider that if we maintained a global timestamp across all clients, then
+requests from each of these clients would need to all be synchronized.
+
+Cluster
+-------
+
+.. note::
+
+   You will probably want to use an odd number of servers; using an even number of servers increases the overhead
+   of distributed consensus while not actually providing any additional fault tolerance as far as obtaining a quorum
+   is concerned. Using more servers leads to improved fault tolerance at the expense of additional overhead incurred
+   in leader election and consensus.
+
+The ``cluster`` block is used to identify the servers which make up a Timelock Service cluster. An example is as
+follows:
+
+   .. code:: yaml
+
+      cluster:
+        localServer: palantir.com:8700
+        servers:
+          - palantir.com:8700
+          - palantir.com:8701
+          - palantir.com:8702
+
+.. list-table::
+   :widths: 5 40
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - localServer
+     - A string following the form ``hostname:port`` which matches the host on which this config exists.
+
+   * - servers
+     - A list of strings following the form ``hostname:port`` identifying the hosts in this Timelock
+       Service cluster. Note that this list must include the ``localServer``.
+
+Algorithm
+---------
+
+We have implemented the consensus algorithm required by the Timelock Server using the Paxos algorithm.
+Currently, this is the only supported implementation, though there may be more in the future.
+We identify an algorithm by its ``type`` field first; algorithms can have additional algorithmic-specific
+customisable parameters as well.
+
+Paxos Configuration
+~~~~~~~~~~~~~~~~~~~
+
+The algorithm has several configurable parameters; all parameters (apart from ``type``) are optional and have
+default values.
+
+   .. code:: yaml
+
+      algorithm:
+        type: paxos
+        paxosDataDir: var/log/paxos
+        sslConfiguration:
+          trustStorePath: var/security/trustStore.jks
+        pingRateMs: 5000
+        maximumWaitBeforeProposalMs: 1000
+        leaderPingResponseWaitMs: 5000
+
+.. list-table::
+   :widths: 5 40
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - type
+     - The type of algorithm to use; currently only ``paxos`` is supported.
+
+   * - paxosDataDir
+     - A path corresponding to the location in which Paxos will store its logs (of accepted promises and learned
+       values) (default: ``var/data/paxos``). The Timelock Server will fail to start if this directory does not
+       exist and cannot be created.
+
+   * - sslConfiguration
+     - Security settings for communication between Timelock Servers, following the
+       `palantir/http-remoting <https://github.com/palantir/http-remoting/blob/develop/ssl-config/src/main/java/com/palantir/remoting1/config/ssl/SslConfiguration.java>`__
+       library (default: no SSL).
```

Comment:
As far as I know, internal products defaulting to having this enabled was to be done via our internal deployability tool.

@hsaraogi probably knows this better than I do, but we can change this later with a PR that changes the defaults if this is the case.